e2e/tcproute_test.go

```diff
@@ -713,17 +714,143 @@ func TestMeshWithMultipleTCPRoutes(t *testing.T) {
 	processGraphAndExpectActions(t, graphBuilder, expectedActions)
 
 	bs, err = theCloud.BackendServices().Get(ctx, removedBS1.Key)
-	if err == nil {
+	if !strings.Contains(err.Error(), "not found") {
 		t.Fatalf("theCloud.BackendServices().Get(_, %v) = %v, nil, want err", removedBS1.Key, bs)
 	}
 
 	hc, err := theCloud.HealthChecks().Get(ctx, removedHC1.Key)
-	if err == nil {
+	if !strings.Contains(err.Error(), "not found") {
 		t.Fatalf("theCloud.HealthChecks().Get(_, %v) = %v, nil, want err", removedHC1, hc)
 	}
 
 	tcpr, err := theCloud.TcpRoutes().Get(ctx, removedTcpr.Key)
-	if err == nil {
+	if !strings.Contains(err.Error(), "not found") {
 		t.Fatalf("theCloud.TcpRoutes().Get(_, %v) = %v, nil, want err", removedTcpr.Key, tcpr)
 	}
 }
+
+func TestTcpRouteDeletion(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+	resUniqueIdPart := "tcpr-deletion"
+	meshURL, meshKey := ensureMesh(ctx, t, resUniqueIdPart+"-mesh")
+
+	t.Cleanup(func() {
+		err := theCloud.Meshes().Delete(ctx, meshKey)
+		t.Logf("theCloud.Meshes().Delete(ctx, %s): %v", meshKey, err)
+	})
+
+	hcs, err := createManyHealthchecks(graphBuilder, 3, resUniqueIdPart)
```

Comment:
future reference -- createManyHealthchecks --> createHealthchecks (Many is redundant).

Reply:
Done.